src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php

```diff
@@ -235,21 +258,28 @@ public static function createFromMovieUrl(
     public static function fromDataTransferObject(MediaItemDataTransferObject $mediaItemDataTransferObject): ?MediaItem
     {
         if (!$mediaItemDataTransferObject->hasExistingMediaItem()) {
-            throw new \BadFunctionCallException('This method can not be used to create a new media item');
+            throw new BadFunctionCallException('This method can not be used to create a new media item');
         }
 
         $mediaItem = $mediaItemDataTransferObject->getMediaItemEntity();
 
-        $mediaItem->title = $mediaItemDataTransferObject->title;
         $mediaItem->folder = $mediaItemDataTransferObject->folder;
         $mediaItem->userId = $mediaItemDataTransferObject->userId;
         $mediaItem->url = $mediaItemDataTransferObject->url;
 
+        foreach ($mediaItemDataTransferObject->translations as $mediaItemTranslationDataTransferObject) {
+            $mediaItemTranslationDataTransferObject->setMediaItem($mediaItem);
+
+            $mediaItem->addTranslation(MediaItemTranslation::fromDataTransferObject($mediaItemTranslationDataTransferObject));
+        }
+
         return $mediaItem;
     }
 
     public function jsonSerialize(): array
     {
+        $locale = BackendLocale::workingLocale();
```

Comment:
not a fan of this, I think a method to set the active language is better with a fallback like this for instance, but now if you would try to use this in the frontend it would fail

Reply:
upi can then also have a method, getActiveTranslation

Reply:
Can you tell me how do you like to see Locale (Backend/Frontend) integrated?

Reply:
@carakas, any example?

Reply:
poke @carakas

Reply:
just add a property for the current locale that isn't persisted to the entity, then add a method to set/change that locale and use that to get the correct translation

don't know the location of a code snippet by heart